pkg/graveler/ref/manager.go

```diff
@@ -387,6 +395,31 @@ func (m *Manager) addCommit(tx db.Tx, repositoryID graveler.RepositoryID, commit
 	return err
 }
 
+func (m *Manager) updateCommitGeneration(tx db.Tx, repositoryID graveler.RepositoryID, commitIDToNode map[graveler.CommitID]CommitNode) error {
+	for len(commitIDToNode) != 0 {
+		command := "WITH updated(id, generation) AS (VALUES "
+		var updatingRows int
+		for commitID, commitNode := range commitIDToNode {
+			if updatingRows != 0 {
+				command += ","
+			}
+			command += fmt.Sprintf("('%s', %d)", commitID, commitNode.generation)
+
+			delete(commitIDToNode, commitID)
```

Comment:
I need this since I'm running the SQL updates in batches so when I iterate again over the dictionary, I want to avoid iterating nodes I've already visited.